pkg/cmd/cli/cmd/loginoptions.go

```diff
@@ -159,6 +159,10 @@ func (o *LoginOptions) getClientConfig() (*restclient.Config, error) {
 				fmt.Fprintln(o.Out)
 			}
 
+		// TLS oversized record which usually means the server only supports "http"
+		case clientcmd.IsTLSOversizedRecord(result.Error()):
+			return nil, fmt.Errorf(clientcmd.GetPrettyMessageFor(result.Error()))
+
 		default:
 			return nil, result.Error()
```

Comment:
I'd rather distinguish between connection-level errors and HTTP status code / body content errors... I don't see why we care about a 404 or non-JSON response here, we really just want to make sure the host resolves and TLS connections can be made